tests/Unit/Amqp/AMQPChannelTest.php

```diff
@@ -0,0 +1,454 @@
+
+        $extendedAmqpChannel->startTransaction();
+    }
+
+    public function testConstructorCorrectlyBridgesTheChannelToTheCreatedChannelBridge(): void
+    {
+        $this->connectionBridge->expects()
+            ->createChannelBridge()
+            ->once()
+            ->andReturn($this->channelBridge);
+
+        new AMQPChannel($this->amqpConnection);
+
+        static::assertSame($this->channelBridge, AmqpBridge::getBridgeChannel($this->amqpChannel));
+    }
+
+    public function testDestructorClosesChannelWhenOpen(): void
+    {
+        $this->amqplibChannel->expects()
+            ->close()
+            ->once();
+
+        $this->amqpChannel = null; // Invoke the destructor synchronously (assuming no reference cycles).
+    }
+
+    public function testDestructorUnregistersChannel(): void
+    {
+        $this->channelBridge->expects()
+            ->unregisterChannel()
+            ->once();
+
+        $this->amqpChannel = null; // Invoke the destructor synchronously (assuming no reference cycles).
+    }
+
+    public function testDestructorDoesNotCloseChannelWhenClosed(): void
+    {
+        $this->amqplibChannel->allows()
+            ->is_open()
+            ->andReturn(false);
+
+        $this->amqplibChannel->expects()
+            ->close()
+            ->never();
+
+        $this->amqpChannel = null; // Invoke the destructor synchronously (assuming no reference cycles).
+    }
+
+    /**
+     * @dataProvider basicRecoverDataProvider
+     */
+    public function testBasicRecoverLogsAttemptAsDebug(bool $requeue): void
+    {
+        $this->amqplibChannel->allows()
+            ->basic_recover($requeue);
+
+        $this->logger->expects()
+            ->debug('AMQPChannel::basicRecover(): Recovery attempt', [
+                'requeue' => $requeue,
+            ])
+            ->once();
+
+        $this->amqpChannel->basicRecover($requeue);
+    }
+
+    /**
+     * @dataProvider basicRecoverDataProvider
+     */
+    public function testBasicRecoverGoesViaAmqplib(bool $requeue): void
+    {
+        $this->amqplibChannel->expects()
+            ->basic_recover($requeue)
+            ->once();
+
+        static::assertTrue($this->amqpChannel->basicRecover($requeue));
+    }
+
+    public function testBasicRecoverHandlesAmqplibExceptionCorrectly(): void
+    {
+        $exception = new AMQPProtocolChannelException(21, 'my text', [1, 2, 3]);
+
+        $this->amqplibChannel->allows()
+            ->basic_recover(true)
+            ->andThrow($exception);
+
+        $this->expectException(AMQPChannelException::class);
+        $this->expectExceptionMessage('AMQPChannel::basicRecover(): Amqplib failure: my text');
+        $this->logger->expects()
+            ->logAmqplibException('AMQPChannel::basicRecover', $exception)
+            ->once();
+
+        $this->amqpChannel->basicRecover();
+    }
+
+    public static function basicRecoverDataProvider(): array
```

Comment:
Why do the 2 data providers need to be static?

Reply:
Because they now do for PHPUnit 10+:

> A data provider method must be public and static. 

https://docs.phpunit.de/en/10.0/writing-tests-for-phpunit.html#data-providers

Reply:
🙏 ✅